lib/atlas/runner.rb

```diff
@@ -13,9 +13,9 @@ class Runner
     # Public: Creates a new Runner.
     #
     # Returns a Runner.
-    def initialize(dataset, graph)
+    def initialize(dataset, graph = nil)
       @dataset = dataset
-      @graph   = graph
+      @graph   = graph || dataset.graph
```

Comment:
I agree with most of your points except this one.

`@graph   = graph || dataset.try(:graph) || GraphBuilder.build`

Which doesn't look that nice to me. I agree that the `graph` is not an attribute of a `FullDataset` but I'm not sure about how this would be an improvement.